### PR TITLE
feat(BMP-828): timestamp with millisecond precision

### PR DESCRIPTION
**Pull request by FilippoRezzonico**
Now timestamps show the time elapsed since January 1st 1970 in milliseconds.

#### Checklist

- [X] your branch will not cause merge conflict with `master`
- [X] run `npm test`
- [X] tests are included
- [ ] the documentation is updated or intregrated accordingly with your changes
- [X] the code will follow the lint rules and style of the project
- [X] you are not committing extraneous files or sensible data

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -7,6 +7,7 @@ github.com/google/uuid v1.1.2/go.mod h1:TIyPZe4MgqvfeYDBFedMoGGpEw/LqOeaOT+nhxU+
 github.com/google/uuid v1.1.3/go.mod h1:TIyPZe4MgqvfeYDBFedMoGGpEw/LqOeaOT+nhxU+yHo=
 github.com/google/uuid v1.1.4/go.mod h1:TIyPZe4MgqvfeYDBFedMoGGpEw/LqOeaOT+nhxU+yHo=
 github.com/google/uuid v1.1.5/go.mod h1:TIyPZe4MgqvfeYDBFedMoGGpEw/LqOeaOT+nhxU+yHo=
+github.com/google/uuid v1.2.0 h1:qJYtXnJRWmpe7m/3XlyhrsLrEURqHRM2kxzoxXqyUDs=
```
Comment (davidebianchi) [inline go.sum:10]: Could you try to run `go mod tidy`. There are several version of the same depenency

File: jsonformatter_test.go [hunk 2]
```diff
@@ -39,43 +39,43 @@ func TestCustomWriter(t *testing.T) {
 				inputLevel:    logrus.TraceLevel,
 				expectedLevel: 10,
 				inputTime:     now,
-				expectedTime:  now.Unix(),
+				expectedTime:  now.UnixNano() / int64(1e6),
```
Comment (davidebianchi) [inline jsonformatter_test.go:42]: We should verify it's the correct time, in this test is very difficult (also if it seems correct)

**Review by davidebianchi — CHANGES_REQUESTED** [review]

File: go.sum (hunk 1 above)
Comment (FilippoRezzonico) [reply go.sum:5]: Done, some dependencies have been removed

File: jsonformatter_test.go (hunk 2 above)
Comment (FilippoRezzonico) [reply jsonformatter_test.go:43]: I have added an assert in the already existing test. Is it ok?

**Review by fredmaggiowski — APPROVED** [review]

File: jsonformatter_test.go [hunk 3]
```diff
@@ -90,9 +91,17 @@ func TestCustomWriter(t *testing.T) {
...
 				assert.Assert(t, err == nil, "failed custom writer writing: %s", err)
-				assert.Equal(t, string(result), fmt.Sprintf("{\"level\":%d,\"msg\":\"test\",\"time\":%d}\n", testCase.expectedLevel, testCase.expectedTime))
+
+				assert.Equal(t, stringResult, fmt.Sprintf("{\"level\":%d,\"msg\":\"test\",\"time\":%d}\n", testCase.expectedLevel, testCase.expectedTime))
+
+				var timestamp int
+				fmt.Sscanf(stringResult,
+					"{\"level\":"+strconv.Itoa(testCase.expectedLevel)+",\"msg\":\"test\",\"time\":%d}",
+					&timestamp)
+				assert.Assert(t, timestamp >= 1e12 && timestamp <= 1e15, "timestamp is not in milliseconds: %d", timestamp)
```
Comment (davidebianchi) [inline jsonformatter_test.go:104]: I don't know if it's ok this check.
Here you could check if the year is correct, for example? Or make a convert and assert time.Now() is in the near to the specific log time (parsing the millisecond)?

Comment (FilippoRezzonico) [reply jsonformatter_test.go:104]: Now I convert the timestamp into date and verify that the year is greater than `2021`

**Pull request merged**